frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditOptionList/OptionListEditor.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { createRef } from 'react';
+import type { IGenericEditComponent } from '@altinn/ux-editor/components/config/componentConfig';
+import type { SelectionComponentType } from '@altinn/ux-editor/types/FormComponent';
+import type { Option } from 'app-shared/types/Option';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCodeListEditor,
+  StudioModal,
+  StudioSpinner,
+  StudioErrorMessage,
+  type CodeListEditorTexts,
+} from '@studio/components';
+import { TableIcon } from '@studio/icons';
+import { useDebounce } from '@studio/hooks';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations/useUpdateOptionListMutation';
+import { useOptionListsQuery } from 'app-shared/hooks/queries/useOptionListsQuery';
+import { useOptionListEditorTexts } from './hooks/useOptionListEditorTexts';
+import { usePreviewContext } from 'app-development/contexts/PreviewContext';
+import classes from './OptionListEditor.module.css';
+import { AUTOSAVE_DEBOUNCE_INTERVAL_MILLISECONDS } from 'app-shared/constants';
+
+type OptionListEditorProps = Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+export function OptionListEditor({ component }: OptionListEditorProps): React.ReactNode {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionsListMap, status } = useOptionListsQuery(org, app);
+
+  switch (status) {
+    case 'pending':
+      return (
+        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
+      );
+    case 'error':
+      return (
+        <StudioErrorMessage>{t('ux_editor.modal_properties_error_message')}</StudioErrorMessage>
+      );
+    case 'success': {
+      return (
+        <OptionListEditorModal
+          optionList={optionsListMap.get(component.optionsId)}
+          component={component}
+        />
+      );
+    }
+  }
+}
+
+type OptionListEditorModalProps = {
+  optionList: Option[];
+} & Pick<IGenericEditComponent<SelectionComponentType>, 'component'>;
+
+function OptionListEditorModal({
+  optionList,
+  component,
```

Comment:
Also nitpicking, but will it work passing only the optionListId instead of the component? Looks lik all we use from the component in the function is the optionListId? 🤔